### ISSUE TITLE
Setting BQ data transfer config disableAutoScheduling to false produces empty scheduleOptions

### ISSUE THREAD

**Issue by musha68k** [hashicorp/terraform-provider-google#13221]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```shell
$ terraform -v
Terraform v1.3.6
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.45.0
+ provider registry.terraform.io/hashicorp/google-beta v4.45.0
```

### Affected Resource(s)

- `google_bigquery_data_transfer_config`

### Terraform Configuration Files

```
resource "google_bigquery_data_transfer_config" "foo" {
  ...
  schedule_options {
     disable_auto_scheduling = false
  }
  ...
}
```

### Debug Output

(target) [apply_google_bigquery_data_transfer_config.log.pgp](https://gist.github.com/musha68k/66f01327a370857f6028335b215837c6/raw/95bc7b9b8249584290d5b8ccf6aa422ed7ed8322/apply_google_bigquery_data_transfer_config.log.pgp)

### Panic Output

None

### Expected Behavior

HTTP PATCH request to [transferConfigs](https://cloud.google.com/bigquery/docs/reference/datatransfer/rest/v1/projects.transferConfigs) endpoint should send corresponding parameter like such:

```
{
  ...
  "scheduleOptions": {
    "disableAutoScheduling": false 
  }
  ...
}
```

### Actual Behavior

Terraform GCP Provider sends request with parameter like such ([empty scheduleOptions](https://cloud.google.com/bigquery/docs/reference/datatransfer/rest/v1/projects.locations.transferConfigs#TransferConfig.ScheduleOptions)):

```
{
  ...
  "scheduleOptions": {}
  ...
}
```

### Steps to Reproduce

`terraform apply`

### Important Factoids

Should be self explanatory.

### References

See links to GCP API documentation in behavior sections above.

**Comment by edwardmedia** [comment]
api does not send `schedule_options.disable_auto_scheduling` back when it was set to false b/262259565

**Comment by nicolas-carlier-ls** [comment]
Hi @edwardmedia, Anything we can do?

**Comment by danekantner** [comment]
Also the api doesn't send disable_auto_scheduling in the call being done, but it does appear to have a method that you can retrieve whether this is already set to true or false

DisableAutoScheduling.get should return this: https://cloud.google.com/dotnet/docs/reference/Google.Cloud.BigQuery.DataTransfer.V1/latest/Google.Cloud.BigQuery.DataTransfer.V1.ScheduleOptions#Google_Cloud_BigQuery_DataTransfer_V1_ScheduleOptions_DisableAutoScheduling

It appears more likely this is a TF provider bug than GCP API bug?

**Comment by ota2000** [comment]
I've submitted a fix for this in GoogleCloudPlatform/magic-modules#18166.

It addresses both sides of the problem:

1. **Request side** (the behavior reported here): `disable_auto_scheduling = false` is the Go zero value, so the expander dropped it and sent an empty `scheduleOptions: {}`. The field now uses `send_empty_value`, so the explicit `false` reaches the API — which also makes it possible to re-enable auto scheduling after it was disabled.
2. **Read side** (the perpetual diff this causes): as noted in https://github.com/hashicorp/terraform-provider-google/issues/13221#issuecomment-1347074387, the API omits `scheduleOptions` from responses when all fields are default, so the block never reached state and every plan re-showed the same in-place update. A custom flattener now preserves the prior state when the API omits the block — restricted to the case where the prior state is itself all-default, so out-of-band changes back to defaults still surface as drift.

Includes an acceptance test that asserts an empty plan after applying `schedule_options { disable_auto_scheduling = false }` and verifies the `false → true → false` round trip.

**Comment by llavoie-trimoz** [comment]
Any updates on when this could be fixed? It's causing a massive amount of logs in multiple environnement for changes that are not really happening.

**Comment by ota2000** [comment]
The fix is ready — it's just not getting reviewed.

GoogleCloudPlatform/magic-modules#18166 has been open since July 3 (5 weeks).
All CI checks are green. It's assigned to a maintainer, the bot has posted five "waiting for review" reminders, and I've pinged twice asking for either a review or a reassignment.
There's still no review on it, so I have no ETA to give you.

If this is hurting your environments, a comment on that PR would carry more weight than anything I can add at this point.

**Comment by llavoie-trimoz** [comment]
@ota2000 Yeah, I saw. I think the issue is that the assigned used does not seems to contribute at the moment. He has taken no action since july. Maybe it's vacation time for them or maybe he was a dev among those who were fired at Google 🤷.
